Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -14,217 +14,297 @@
-    return StubReturn();
+    return [self orderedSetWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithOrderedSet:set range:range copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithSet:set] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[self alloc] initWithSet:set copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (array == nil) {
+        return [self init];
+    }
+
+    if (NSMaxRange(range) > [array count]) {
+        [NSException raise:NSInvalidArgumentException
+                    format:@"The given range length is greater than the array length: %u.", [array count]];
+    }
+
+    if (self = [self init]) {
+        NSUInteger start = range.location;
+        NSUInteger end = NSMaxRange(range);
+
+        if (flag) {
+            for (NSUInteger i = start; i < end; i++) {
+                id obj = [[array objectAtIndex:i] copy];
+                [self _insertObject:obj];
+                [obj release];
+            }
+        } else {
+            for (NSUInteger i = start; i < end; i++) {
+                [self _insertObject:[array objectAtIndex:i]];
+            }
+        }
+    }
+    return self;
+}
+
+- (void)_insertObject:(id)object {
+    if ([self containsObject:object]) {
+        return;
+    }
+
+    [_arrayContainer addObject:object];
+    [_setContainer addObject:object];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (NSUInteger)count {
+    return [_arrayContainer count];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSArray*)array {
+    return [[[_NSProxyOrderedSetArray alloc] initWithOrderedSet:self] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
+- (NSSet*)set {
+    return [[_setContainer copy] autorelease];
```

Comment:
Oh, I see why the proxy array approach was taken. 
Interesting: we need to effectively strip the mutability without losing the passthrough.
Do you think we can just return the internal mutable container and hope nobody decides to manipulate it?